hack/ensure-golangci-lint.sh

```diff
@@ -1,22 +1,427 @@
-#!/bin/sh
-
-set -eux
-
-IS_CONTAINER="${IS_CONTAINER:-false}"
-CONTAINER_RUNTIME="${CONTAINER_RUNTIME:-podman}"
-
-if [ "${IS_CONTAINER}" != "false" ]; then
-    export XDG_CACHE_HOME=/tmp/.cache
-    mkdir /tmp/unit
-    cp -r . /tmp/unit
-    cd /tmp/unit
-    make lint
-else
-    "${CONTAINER_RUNTIME}" run --rm \
-        --env IS_CONTAINER=TRUE \
-        --volume "${PWD}:/workdir:ro,z" \
-        --entrypoint sh \
-        --workdir /workdir \
-        docker.io/golang:1.19 \
-        /workdir/hack/ensure-golangci-lint.sh
+#!/usr/bin/env bash
+
+# Copyright 2023 The Metal3 Authors.
```

Comment:
I do not think this is right. We don't get the copyright just because we made some small changes. To make things worse, we here (and CAPI also) changed the license to Apache 2, whereas the original is GNU GPL v3. I'm no lawyer but I know GPL is stricter than Apache so this could well be a violation of the original license.

Reply:
I'm especially concerned about this part from the [GPL license](https://github.com/golangci/golangci-lint/blob/master/LICENSE):

    c) You must license the entire work, as a whole, under this
    License to anyone who comes into possession of a copy.  This
    License will therefore apply, along with any applicable section 7
    additional terms, to the whole of the work, and all its parts,
    regardless of how they are packaged.  This License gives no
    permission to license the work in any other way, but it does not
    invalidate such permission if you have separately received it.

Reply:
@lentzi90 if CAPI can live with it, I see no point of making this a critical issue for CAPM3, CAPI has copied it and put a different licence, not us, we have copied CAPI and adapted it which is clearly mentioned. I am no lawyer as well.

Reply:
If this is such a big deal , then I can put back the Kubernetes authors Copyright although we decided the other way around offline.

Reply:
We did not decide to put Metal3 to non-original work. If you copy stuff, you keep the copyright etc as is. For original work we shouldn't be putting Kubernetes Authors as they did not author it, but Metal3 Authors did.

As for the GPL change in CAPI, I suggest we ask them about it. GPL is quite shunned upon for many businesses, and I would not like us to take it in if GPL.

Reply:
The current file is not a full original copy. It has small change. How do you decide the copyright in such case? I cant put a GPL license here. The most I can do is I put CAPI's original copyright notice. Anyways, I will check the license with someone who knows more.

Reply:
Is this solved?

Reply:
Nothing to solve on our side, CAPI has been informed and once CAPI reflects back the correct licensing, we shall do the same on our side.

Reply:
Well that is not how licenses work.

Reply:
Changed the file